### DISCUSSION TITLE
RSGI support

### DISCUSSION THREAD

**Discussion by KaliszS** [encode/starlette#2287] [Ideas] [answered]
Hello. Is that possible to add RSGI support? It would be awesome to use granian with RSGI interface instead of uvivorn with ASGI.

granian: https://github.com/emmett-framework/granian
benchmarks: https://github.com/emmett-framework/granian/blob/master/benchmarks/README.md
rsgi specification: https://github.com/emmett-framework/granian/blob/master/docs/spec/RSGI.md

**Comment by Kludex** [comment, marked as answer]
I've already talked to @gi0baro about this.

I don't think it makes sense for Starlette to support another interface. It's just that Starlette is built in blocks, every block being a self contained ASGI application.

**Comment by conradogarciaberrotaran** [comment]
Hola!
does this mean that Starlette won't ever support Granian?

**Reply by gi0baro** [reply]
@conradogarciaberrotaran that's not the way to look at this. Granian supports several different application interfaces: you can use Granian with every ASGI framework/application, and thus also Starlette.

RSGI is just another protocol which has pros and cons when compared to ASGI, but it's definitely a choice of the framework maintainers which protocols they want to support :)

**Reply by conradogarciaberrotaran** [reply]
Got it!
Thanks for the quick response and for building Granian!
